cmd/dlt/accountroles/cmd.go

```diff
@@ -0,0 +1,250 @@
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/interactive"
+	"github.com/openshift/rosa/pkg/interactive/confirm"
+	"github.com/openshift/rosa/pkg/logging"
+	"github.com/openshift/rosa/pkg/ocm"
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+	"github.com/spf13/cobra"
+)
+
+var modes []string = []string{"auto", "manual"}
+
+var args struct {
+	roleName string
+	mode     string
+	prefix   string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "account-roles",
+	Aliases: []string{"accountroles", "accountrole", "account-role"},
+	Short:   "Delete Account Roles",
+	Long:    "Cleans up account roles from the current AWS account.",
+	Example: `  # Delete Account roles"
+  rosa delete account-roles [-r role-name | -p prefix]`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVar(
+		&args.mode,
+		"mode",
+		modes[0],
+		"How to perform the operation. Valid options are:\n"+
+			"auto: Account roles will be deleted automatically using the current AWS account\n"+
+			"manual: Command to delete the account roles will be output which can be used to delete manually",
+	)
+	Cmd.RegisterFlagCompletionFunc("mode", modeCompletion)
+
+	flags.StringVarP(
+		&args.roleName,
+		"role-name",
+		"r",
+		"",
+		"Account role name to be deleted.",
+	)
+
+	flags.StringVarP(
+		&args.prefix,
+		"prefix",
+		"p",
+		"",
+		"Prefix of the account roles to be deleted.",
+	)
+
+	confirm.AddFlag(flags)
+}
+
+func modeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return modes, cobra.ShellCompDirectiveDefault
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	if cmd.Flags().Changed("prefix") && cmd.Flags().Changed("role-name") {
+		reporter.Errorf("Provide either prefix or role-name not both")
+		os.Exit(1)
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	env, err := ocm.GetEnv()
+	if err != nil {
+		reporter.Errorf("Error getting environment %s", err)
+		os.Exit(1)
+	}
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+	clusters, err := ocmClient.GetClusters(creator, 1000)
```

Comment:
Any chance we can load _all_ clusters here? It's unlikely that any given customer has 1000 cluster, but I'd rather not mislead.